CMakeLists.txt.in

```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
```

Comment:
I think you mentioned that gtest fails to build w/ cmake-2.8? What is the real requirement? Do we have any hint in gtest readme?

Reply:
Looks like the requirement is for the CMake version to be at least 3.0:
https://github.com/google/googletest/issues/1786

Reply:
Have you tried with the `release-1.8.0` tag?

Reply:
Indeed, the release-1.8.0 tag version allows to configure the build with CMake 2.8.12 as well.

Reply:
However, the release-1.8.0 does not compile - seems to be an internal bug in GTest. We will have to switch to CMake3 after all.

Reply:
What is the compile error?  For other projects, like intel-vaapi-driver, libva-utils and libyami release-1.8.0 compiles fine in multiple environments... of course, those projects use automake, though.

Reply:
@vshampor,

JFI: release-1.8.x is pre-C++11 and uses a technic of functions overload bunch (_ElementsAre_ matcher is eg.) instead of variadic - this leads to slower compilation.

Reply:
See http://mediasdk.intel.com/buildbot/#/builders/6/builds/1936 for errors. This is reproduced locally on my machine running CentOS 7.4/CMake 2.8.12 as well.

Reply:
I see, it fails in gmock.  I've never tried compiling gmock, sorry :/

Reply:
@vshampor looks like the `error: ‘mock.MockCallObj::gmock_dlclose’ does not have class type` errors are due to incorrect gmock usage.  https://github.com/google/googletest/blob/release-1.8.0/googlemock/test/gmock-spec-builders_test.cc#L225 has examples of how the MOCK_METHODn is used with EXPECT_CALL.

Perhaps this will help improve your forked PR.

Reply:
@uartie, @dmitry-gurulev  : we are working to clean up branches which we have in this repository and started to prohibit PR from within major repo. This PR is one of these that's why @vshampor closed it and opened new one: #856. I apologize for the inconveniences, but could you, please, continue discussion in #856?